app/src/main/java/com/example/trempel/ui/pdp/RecentlyViewedAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.trempel.ui.pdp
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.trempel.databinding.RecentlyViewedItemBinding
+import com.example.trempel.db.RecentlyViewed
+
+internal class RecentlyViewedAdapter :
+    ListAdapter<RecentlyViewed, RecentlyViewedAdapter.RecentlyViewHolder>(RecentlyViewedComparator()) {
+
+    var itemList = mutableListOf<RecentlyViewed>()
+    var onItemClicked: ((Int) -> Unit)? = null
+
+    class RecentlyViewHolder(
+        val binding: RecentlyViewedItemBinding
+    ) : RecyclerView.ViewHolder(binding.root)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecentlyViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+        val iteView = RecentlyViewedItemBinding.inflate(layoutInflater, parent, false)
+        return RecentlyViewHolder(iteView)
+    }
+
+    override fun onBindViewHolder(holder: RecentlyViewHolder, position: Int) {
+        holder.binding.recentlyView = itemList[position]
+        holder.binding.containerRecentlyViewItem.setOnClickListener {
+            onItemClicked?.invoke(itemList[position].id)
+        }
+    }
+
+    override fun getItemCount(): Int {
+        return itemList.size
+    }
+}
+
+class RecentlyViewedComparator : DiffUtil.ItemCallback<RecentlyViewed>() {
```

Comment:
can we make it `internal` and move it to another file?

Reply:
done